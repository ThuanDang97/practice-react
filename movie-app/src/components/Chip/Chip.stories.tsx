import { Story } from '@storybook/react';
import Chip from './Chip';
import { TProps } from './Chip';

export default {
  title: 'Components/Chip',
  component: Chip,
};

const Template: Story<TProps> = (args): JSX.Element => <Chip {...args} />;

export const Basic: Story<TProps> = Template.bind({});
Basic.args = {
  chip: {
    id: 1,
    name: 'Chip',
    actived: false,
  },
};

export const Actived: Story<TProps> = Template.bind({});
Actived.args = {
  chip: {
    id: 1,
    name: 'Chip',
    actived: true,
  },
};
