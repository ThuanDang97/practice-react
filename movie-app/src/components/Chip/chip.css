.list-chip {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  gap: 10px;
}

.chip-element {
  padding: 10px 18px;
  border: black 1px solid;
  border-radius: 15px;
  text-align: center;
  cursor: pointer;
  color: black;
  line-height: 15px;
}

.chip-element:hover {
  background-color: rgba(1, 180, 228, 1);
  color: #fff;
  border-color: rgba(1, 180, 228, 1);
}

.active {
  background-color: rgba(1, 180, 228, 1);
  color: #fff;
  border-color: rgba(1, 180, 228, 1);
}
