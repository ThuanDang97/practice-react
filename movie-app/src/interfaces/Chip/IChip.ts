export interface IChip {
  id: number;
  name: string;
  actived: boolean;
}
