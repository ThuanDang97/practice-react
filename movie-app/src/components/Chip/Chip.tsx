import { memo, MouseEvent } from 'react';
import { IChip } from '../../interfaces/Chip/IChip';
import './chip.css';

export type TProps = {
  chip: IChip;
  onHandleClick: (e: MouseEvent<HTMLButtonElement>) => void;
};

const Chip = (props: TProps) => {
  const { chip, onHandleClick } = props;

  return (
    <button
      value={chip?.id}
      className={chip?.actived ? 'chip-element active' : 'chip-element'}
      onClick={onHandleClick}
    >
      {chip?.name}
    </button>
  );
};

export default memo(Chip);
