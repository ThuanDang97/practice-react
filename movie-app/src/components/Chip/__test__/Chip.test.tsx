import { screen, render } from '@testing-library/react';
import { setup } from '../../../support/Setup';
import Chip, { TProps } from '../Chip';

describe('Test Chip component', () => {
  const defaultProps: TProps = {
    chip: {
      id: 1,
      name: 'Fantastic',
      actived: false,
    },
    onHandleClick: jest.fn(),
  };

  const customProps: TProps = {
    chip: {
      id: 1,
      name: 'Fantastic',
      actived: true,
    },
    onHandleClick: jest.fn(),
  };

  test('Component Chip matches DOM Snapshot', () => {
    const component = <Chip {...defaultProps} />;
    const { container } = render(component);

    expect(container).toMatchSnapshot();
  });

  test('Component Chip should render correctly', () => {
    setup(<Chip {...defaultProps} />);

    expect(screen.getByRole('button')).toHaveTextContent('Fantastic');
  });

  test('Component Chip should have class active when actived is true', () => {
    setup(<Chip {...customProps} />);

    expect(screen.getByRole('button').classList).toContain('active');
  });

  test('Component Chip should called handleClick', async () => {
    const { user } = setup(<Chip {...defaultProps} />);

    await user.click(screen.getByRole('button', { name: /Fantastic/i }));

    expect(defaultProps.onHandleClick).toHaveBeenCalled();
  });
});
