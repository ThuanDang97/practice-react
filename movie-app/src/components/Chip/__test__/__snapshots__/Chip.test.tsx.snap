// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Chip component Component Chip matches DOM Snapshot 1`] = `
<div>
  <button
    class="chip-element"
    value="1"
  >
    Fantastic
  </button>
</div>
`;
